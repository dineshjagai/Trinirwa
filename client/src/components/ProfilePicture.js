/* eslint-disable react/destructuring-assignment */
import React, { useContext } from 'react';
import './Profile.css';
import LockOpenIcon from '@material-ui/icons/LockOpen';
import DialogPassword from './DialogPassword';
import DialogPasswordChange from './DialogPasswordChange';
import ScrollDialog from './DisplayerDialog';
import { getBlockedFollowers, unBlockUser } from './Module';
import idContext from './Context';

export default function Profile(props) {
  const user = useContext(idContext);
  const handleUnblock = (username) => {
    unBlockUser(user, username).then((res) => {
      console.log(res.message);
      getBlockedFollowers(user).then((result) => {
        console.log(result.data.friends);
      });
    });
  };

  return (
    <div className="container-profile">
      <div className="pictures">
        <img src={props.data.profile_picture} alt="" id="overlay" />
        <img src={props.data.profile_picture} alt="" id="originalDiv" />
      </div>
      <br />
      <div className="username">
        <span id="username">{props.data.username}</span>
      </div>
      <div className="deleteButton">
        <ScrollDialog
          secondary="blocked"
          title="My Blocked Users"
          secondTitle="Blocked Users"
          getFunction={getBlockedFollowers}
          Icon={LockOpenIcon}
          handle={handleUnblock}
          iconText="Unblock User"
        />
        <DialogPassword id={1220} />
        <DialogPasswordChange id={1220} />
      </div>
    </div>
  );
}

// export default function Profile(props) {
//   return (
//     <div className="container-profile">
//       <div className="pictures">
//         <img src={props.data.profile_picture} alt="" id="overlay" />
//         <img src={props.data.profile_picture} alt="" id="originalDiv" />
//       </div>
//       <br />
//       <div className="username">
//         <span id="username">{props.data.username}</span>
//       </div>
//       <div className="deleteButton">
//         <DialogPassword id={props.id} />
//         <DialogPasswordChange id={props.id} />
//       </div>
//     </div>
//   );
// }
