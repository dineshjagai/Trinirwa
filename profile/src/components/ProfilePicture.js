import React, { Component, Profiler } from 'react';
import axios from 'axios';
import './Profile.css';

export default function Profile(props){
    let input = props.picture;
    console.log(props.picture);
    return(
        <div className= "pictures">
            <img className = "pictures" src = {props.picture} id = "overlay"></img>
            <img className="pictures" src = {props.picture} id = "originalDiv"></img>
        </div>
    );
}