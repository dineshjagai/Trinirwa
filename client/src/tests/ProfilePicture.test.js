import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import ProfilePicture from '../components/ProfilePicture';

test('Test ProfilePicture snapshot', () => {
  const component = renderer.create(<ProfilePicture />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
