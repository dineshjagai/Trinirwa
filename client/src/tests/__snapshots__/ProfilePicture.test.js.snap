// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfilePicture snapshot 1`] = `
<div
  className="container-profile"
>
  <div
    className="pictures"
  >
    <img
      alt=""
      id="overlay"
      src="/api/viewFile/logo512.png"
    />
    <img
      alt=""
      id="originalDiv"
      src="/api/viewFile/logo512.png"
    />
  </div>
  <br />
  <div
    className="username"
  >
    <span
      id="username"
    >
      isimbib
    </span>
  </div>
  <div
    className="deleteButton"
  >
    <div>
      <button
        className="btn"
        id="delete"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "inherit",
            "border": "none",
            "color": "#00695c",
          }
        }
        type="button"
      >
        My Blocked Users
      </button>
    </div>
    <div>
      <button
        id="delete"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "inherit",
            "border": "none",
            "color": "#00695c",
          }
        }
        type="button"
      >
        Delete account
      </button>
    </div>
    <div>
      <button
        id="change"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "inherit",
            "border": "none",
            "color": "#00695c",
          }
        }
        type="button"
      >
        Change password
      </button>
    </div>
  </div>
</div>
`;
